src/App.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState } from 'react';
+
+import Page from './Page';
+
+function App() {
+  const [state, setState] = useState({
+    inputValue: '',
+    todoItems: [],
+
+  });
+
+  const { inputValue, todoItems } = state;
+
+  function handleDoneButtonClick(selectedItem) {
+    const newTodoItems = todoItems.filter((item) => item.id !== selectedItem.id);
+    setState({ inputValue, todoItems: newTodoItems });
+  }
+
+  function handleInputSubmit(event) {
+    event.preventDefault();
+    if (!inputValue.trim()) return;
+    setState({ inputValue: '', todoItems: [...todoItems, { todo: inputValue, id: new Date() }] });
+  }
+
+  function handleInputChange(event) {
+    setState({ inputValue: event.target.value, todoItems });
+  }
```

Comment:
오히려 더 불편해진게 아닌가 라는 고민을 해보신게 너무 좋습니다 :) 장단점이 있지만, 제가 말씀드린 부분은 하나의 상태로 관리가 가능하다는 것과, es6 spread 문법을 이용하면 고민하신 부분들이 조금 수월하게 해결된다는 점때문이었습니다 ㅎㅎ
```suggestion
  function handleInputChange(event) {
    setState({
      ...state,
      inputValue: event.target.value
    });
  }
```
굳이 state를 한번 더 써줘는거면 똑같이 불편하다라고 생각할 수 있지만, 이렇게 처리함으로써 이 함수에서 신경쓸 필요 없는 다른 값들은 그대로 가져오고, 신경써야하는 inputValue값만 event.target.value로 업데이트 해줄 수 있다는 점이 장점이지 않을까요? :)